packages/editor-ui/src/components/mixins/workflowHelpers.ts

```diff
@@ -326,10 +327,17 @@ export const workflowHelpers = mixins(
 					const nodeParameters = NodeHelpers.getNodeParameters(nodeType.properties, node.parameters, false, false, node);
 					nodeData.parameters = nodeParameters !== null ? nodeParameters : {};
 
+					const fullAccess = node.type === HTTP_REQUEST_NODE_TYPE && node.typeVersion === 2;
+
 					// Add the node credentials if there are some set and if they should be displayed
 					if (node.credentials !== undefined && nodeType.credentials !== undefined) {
 						const saveCredenetials: INodeCredentials = {};
 						for (const nodeCredentialTypeName of Object.keys(node.credentials)) {
+							if (fullAccess) {
+								saveCredenetials[nodeCredentialTypeName] = node.credentials[nodeCredentialTypeName];
+								continue;
+							}
```

Comment:
why is this needed?

Reply:
To prevent cred ref from being lost. `find()` on the next line will never find anything for HTTPRN because this is the only node able to access cred types not defined on itself.

https://linear.app/n8n/issue/N8N-3354#comment-0ce9d056